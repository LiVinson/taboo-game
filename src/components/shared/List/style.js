import styled from "styled-components"

export const StyledList = styled.ul`
        list-style-type:none;
`