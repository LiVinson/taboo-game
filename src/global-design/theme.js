const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
