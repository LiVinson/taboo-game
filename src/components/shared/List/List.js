import React from "react"
import {StyledList} from "./style"

const List = ({children}) => (
    <StyledList>
        {children}
    </StyledList>
)

export default List