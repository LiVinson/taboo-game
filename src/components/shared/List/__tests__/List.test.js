import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'
import theme from '../../../../global-design/theme'
import List from "../List"

test('List renders children correctly ', () => {
	const list = ['item1', 'item2', 'item3']
	const wrapper = renderer
		.create(
			<ThemeProvider theme={theme}>
				<List>
					{list.map((item, index) => (
						<li key={index}>{item}</li>
					))}
				</List>
			</ThemeProvider>
		)
		.toJSON()

	expect(wrapper).toMatchSnapshot()
})
