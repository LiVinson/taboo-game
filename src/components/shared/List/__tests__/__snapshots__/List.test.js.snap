// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders children correctly  1`] = `
.c0 {
  list-style-type: none;
}

<ul
  className="c0"
>
  <li>
    item1
  </li>
  <li>
    item2
  </li>
  <li>
    item3
  </li>
</ul>
`;
